Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs

```diff
@@ -1464,18 +1438,18 @@ private static T FindImmediateChild<T>(HierarchyNode parent)
                 }
 
                 var files = AllVisibleDescendants.OfType<PythonFileNode>().Select(f => f.Url).ToArray();
+                var fileSet = new Lazy<HashSet<string>>(() => new HashSet<string>(files, StringComparer.OrdinalIgnoreCase));
 
                 log?.WriteLine($"Project includes files:{Environment.NewLine}    {string.Join(Environment.NewLine + "    ", files)}");
 
-                if (defAnalyzer != null) {
-                    foreach (var f in files) {
-                        var entry = defAnalyzer.GetAnalysisEntryFromPath(f);
-                        if (entry != null) {
-                            log?.WriteLine($"Unloading {f} from default analyzer");
-                            await defAnalyzer.UnloadFileAsync(entry);
-                            foreach (var b in entry.TryGetBufferParser().AllBuffers) {
+                foreach (var existing in _services.Python.GetActiveSharedAnalyzers().Select(kv => kv.Value).Where(v => !v.IsDisposed)) {
```

Comment:
Do you need to try/catch ObjectDisposedException? Can it become disposed while this loop is executing?